astropy/io/ascii/cds.py

```diff
@@ -164,14 +207,344 @@ def get_cols(self, lines):
+            Byte-by-byte Description of file: table.dat
+            --------------------------------------------------------------------------------
+             Bytes Format Units  Label     Explanations
+            --------------------------------------------------------------------------------
+             1- 8   A8     ---    names    Description of names
+            10-14   E5.1   ---    e       [0.0/0.01]? Description of e
+            16-18   F3.0   ---    d       ? Description of d
+            20-26   E7.1   ---    s       [-9e+34/2.0] Description of s
+            28-30   I3     ---    i       [-30/67] Description of i
+            32-34   F3.1   ---    sameF   [5.0/5.0] Description of sameF
+            36-37   I2     ---    sameI   [20] Description of sameI
+
+            --------------------------------------------------------------------------------
+        """
+        # get column widths.
+        vals_list = []
+        col_str_iters = self.data.str_vals()
+        for vals in zip(*col_str_iters):
+            vals_list.append(vals)
+
+        for i, col in enumerate(self.cols):
+            col.width = max([len(vals[i]) for vals in vals_list])
+            if self.start_line is not None:
+                col.width = max(col.width, len(col.info.name))
+        widths = [col.width for col in self.cols]
+
+        startb = 1
+        # set default width of Start Byte, End Byte, Format and
+        # Label columns in the ByteByByte table.
+        sz = [0, 0, 1, 7]
+        l = len(str(sum(widths)))
+        if l > sz[0]:
+            sz[0] = l
+            sz[1] = l
+        for column in self.cols:
+            if len(column.name) > sz[3]:
+                sz[3] = len(column.name)
+
+        buff = ""
+        maxDescripSize = 16
+        nsplit = sum(sz) + 16
+        # format string for Start Byte and End Byte
+        fmtb = "{0:" + str(sz[0]) + "d}-{1:" + str(sz[1]) + "d} {2:" + str(sz[2]) + "s}"
+
+        bbb = Table(names=['Bytes', 'Format', 'Units', 'Label', 'Explanations'],
+                    dtype=[str]*5)
+
+        for i, col in enumerate(self.cols):
+            # check if column is MaskedColumn
+            col.hasNull = isinstance(col, MaskedColumn)
+
+            # set CDSColumn type, size and format.
+            if np.issubdtype(col.dtype, np.int):
+                # integer formatter
+                self.__set_column_val_limits(col)
+                col.meta.size = len(str(col.max))
+                l = len(str(col.min))
+                if col.meta.size < l: col.meta.size = l
+                col.fortran_format = "I" + str(col.meta.size)
+                col.format = ">" + col.fortran_format[1:]
+
+            elif np.issubdtype(col.dtype, np.float):
+                # float formatter
+                self.__set_column_val_limits(col)
+                self.columnFloatFormatter(col)
+
+            else:
+                # string formatter
+                if col.hasNull:
+                    mcol = col
```

Comment:
Nope, copy of the column is not needed. The temporary `coltmp` is only to fill empty string in place of Null values for String columns, so that `col.dtype.str` returns proper column dtype.